ivy/functional/frontends/numpy/ndarray/ndarray.py

```diff
@@ -154,6 +154,14 @@ def cumsum(self, *, axis=None, dtype=None, out=None):
             dtype=dtype,
             out=out,
         )
+    
+    def diagonal(self, *, offset=0, axis1=0, axis2=1):
+        return np_frontend.diagonal(
+            self.data,
```

Comment:
```suggestion
            self._ivyArray,
```
Here the recent refactoring change to use `_ivyArray` instead of `data`